vortex-serde/src/layouts/pruning.rs

```diff
@@ -0,0 +1,538 @@
+        let eq_expr = Arc::new(BinaryExpr::new(
+            Arc::new(Column::new(column.clone())),
+            Operator::Eq,
+            Arc::new(Column::new(other_col.clone())),
+        )) as _;
+
+        let (converted, refs) = convert_to_pruning_expression(&eq_expr);
+        assert_eq!(
+            refs,
+            HashMap::from_iter([
+                (
+                    stat_column_name(&column, Stat::Min),
+                    ColumnStat::new(column.clone(), Stat::Min)
+                ),
+                (
+                    stat_column_name(&column, Stat::Max),
+                    ColumnStat::new(column.clone(), Stat::Max)
+                ),
+                (
+                    stat_column_name(&other_col, Stat::Min),
+                    ColumnStat::new(other_col.clone(), Stat::Min)
+                ),
+                (
+                    stat_column_name(&other_col, Stat::Max),
+                    ColumnStat::new(other_col.clone(), Stat::Max)
+                )
+            ])
+        );
+        let expected_expr: Arc<dyn VortexExpr> = Arc::new(BinaryExpr::new(
+            Arc::new(BinaryExpr::new(
+                Arc::new(Column::new(stat_column_name(&column, Stat::Min))),
+                Operator::Lte,
+                Arc::new(Column::new(stat_column_name(&other_col, Stat::Max))),
+            )),
+            Operator::And,
+            Arc::new(BinaryExpr::new(
+                Arc::new(Column::new(stat_column_name(&other_col, Stat::Min))),
+                Operator::Lte,
+                Arc::new(Column::new(stat_column_name(&column, Stat::Max))),
+            )),
+        ));
+        assert_eq!(*converted, *expected_expr.as_any());
+    }
+
+    #[test]
+    pub fn pruning_not_equals_column() {
+        let column = Field::from("a");
+        let other_col = Field::from("b");
+        let not_eq_expr = Arc::new(BinaryExpr::new(
+            Arc::new(Column::new(column.clone())),
+            Operator::NotEq,
+            Arc::new(Column::new(other_col.clone())),
+        )) as _;
+
+        let (converted, refs) = convert_to_pruning_expression(&not_eq_expr);
+        assert_eq!(
+            refs,
+            HashMap::from_iter([
+                (
+                    stat_column_name(&column, Stat::Min),
+                    ColumnStat::new(column.clone(), Stat::Min)
+                ),
+                (
+                    stat_column_name(&column, Stat::Max),
+                    ColumnStat::new(column.clone(), Stat::Max)
+                ),
+                (
+                    stat_column_name(&other_col, Stat::Min),
+                    ColumnStat::new(other_col.clone(), Stat::Min)
+                ),
+                (
+                    stat_column_name(&other_col, Stat::Max),
+                    ColumnStat::new(other_col.clone(), Stat::Max)
+                )
+            ])
+        );
+        let expected_expr: Arc<dyn VortexExpr> = Arc::new(BinaryExpr::new(
```

Comment:
is this correct? shouldn't a ≠ b be

a.min >= b.max || a.max <= b.min

?

Reply:
I thought so as well but having range 0-9 doesn't mean that all values are covered. For inequality you can only prune if you can confirm the value exists in the dataset